cleverhans/utils_tf.py

```diff
@@ -200,10 +199,12 @@ def model_eval(sess, x, y, model, X_test, Y_test, args=None):
 
             # The last batch may be smaller than all others, so we need to
             # account for variable batch size here
-            accuracy += cur_batch_size * acc_value.eval(
+            cur_acc = acc_value.eval(
                 feed_dict={x: X_test[start:end],
                            y: Y_test[start:end],
                            keras.backend.learning_phase(): 0})
+
+            accuracy += (cur_batch_size * cur_acc).mean()
```

Comment:
@goodfeli 

1.2 Version of keras had mean being called inside categorical_accuracy:

```
def categorical_accuracy(y_true, y_pred):
    '''Calculates the mean accuracy rate across all predictions for
    multiclass classification problems.
    '''
    return K.mean(K.equal(K.argmax(y_true, axis=-1),
                  K.argmax(y_pred, axis=-1)))
```


Which returned a scalar that that was then being aggregated by your code for each batch, before calculating the total avg.

However, 2.0 version removed the Mean calculation from inside the method, and added a K.cast wrapping the call, which results in categorical_accuracy returning an array and, therefore, messing with the calculation when the batch size was smaller than the pre-defined value.
```
def categorical_accuracy(y_true, y_pred):
    return K.cast(K.equal(K.argmax(y_true, axis=-1),
                          K.argmax(y_pred, axis=-1)),
                  K.floatx())
```
Let me know your thoughts on my approach. 
Thanks